### ISSUE TITLE
Possible derefence of null pointer

### ISSUE THREAD

**Issue by Bbulatov** [pgbouncer/pgbouncer#1212]
Hello!
During the static analysis was found possible mistake in code:
1) Return value of a function 'pktbuf_dynamic' (client.c:990) is dereferenced at client.c:993 without checking for NULL, but it is usually checked for this function
![изображение](https://github.com/user-attachments/assets/470f4c95-aa54-4268-962d-cab0017a7ec3)
Please clarify whether verification is required in this case? Do you need to make any changes/corrections to the code?

**Comment by eulerto** [comment]
Good catch! That's an oversight in the commit 6a391797e936aa32fc8ad179c59a7269190b2330.

@JelteF While inspecting this code path, I think you forgot a `return false` into the `if (!res)` block.
